src/App.js

```diff
@@ -96,6 +98,39 @@ type RouteInformation = {
   equipmentInfoId: ?string,
 };
 
+function getRouteInformation(props: Props): ?RouteInformation {
+  const location = props.location;
+  const allowedResourceNames = ['nodes', 'categories', 'search'];
+  const match = location.pathname.match(/(?:\/beta)?\/?(?:(-?\w+)(?:\/([-\w\d]+)(?:\/([-\w\d]+)(?:\/([-\w\d]+))?)?)?)?/i);
+  if (match) {
+    if (match[1] && !includes(allowedResourceNames, match[1])) return null;
+    return {
+      featureId: match[1] === 'nodes' ? match[2] : null,
+      equipmentInfoId: (match[1] === 'nodes' && match[3] === 'equipment') ? match[4] : null,
+      category: match[1] === 'categories' ? match[2] : null,
+      searchQuery: match[1] === 'search' ? parseQueryParams(location.search).q : null,
+      isEditMode: (match[3] === 'edit'),
+    };
+  }
+  return null;
+}
+
+
+function getFeatureIdFromProps(props: Props): ?string {
+  const { featureId } = getRouteInformation(props) || {};
+  return featureId ? String(featureId) : null;
+}
+
+
+function featureIdHasChanged(newProps: Props, prevState: State) {
+  const result = String(getFeatureIdFromProps(newProps)) !== String(prevState.featureId);
```

Comment:
-1: getFeatureIdFromProps already returns a String, please remove the conversions - also `String(null)` is "null"